src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -3595,7 +3595,21 @@ private void MUL_Handler(Instruction instruction)
 
             if (opdata1.IsInteger && opdata2.IsInteger)
             {
-                opdata2 = StackValue.BuildInt(opdata1.IntegerValue * opdata2.IntegerValue);
+                long val;
+                try
+                {
+                    val = checked(opdata1.IntegerValue * opdata2.IntegerValue);
+                }
+                catch (OverflowException)
+                {
+                    opdata2 = StackValue.BuildInt(opdata1.IntegerValue * opdata2.IntegerValue);
+                    runtimeCore.RuntimeStatus.LogWarning(WarningID.Overflow, Resources.OverflowWarning);
```

Comment:
The PR checks are going to fail as I haven't added definitions for `WarningID.Overflow` and `Resources.OverflowWarning` but hopefully this gives some idea.